test/test-nautilus.sh

```diff
@@ -0,0 +1,223 @@
+  $ECHO "$RED[!] failed to compile nautilus custom mutator"
+  exit 1
+}
+
+LIB_PATH="${CUSTOM_MUTATOR_PATH}/target/release/liblibafl_nautilus.${DLL}"
+
+if [ ! -f "$LIB_PATH" ]; then
+    $ECHO "$RED[!] liblibafl_nautilus.${DLL} not found at $LIB_PATH"
+    exit 1
+fi
+
+$ECHO "$BLUE[*] Running libafl_base mutator unit tests"
+(cd "$CUSTOM_MUTATOR_PATH/../libafl_base" && cargo test --features="mutator") || exit 1
+
+$ECHO "$BLUE[*] Running libafl_nautilus unit tests"
+(cd "$CUSTOM_MUTATOR_PATH" && cargo test) || exit 1
+
+LIBAFL_BASE_PATH=$(cd $(pwd)/../custom_mutators/libafl_base;pwd)
+$ECHO "$BLUE[*] Running libafl_base unit tests"
+(cd "$LIBAFL_BASE_PATH" && cargo test --features=mutator) || exit 1
+
+  # Create the vulnerable program
+  cat > test-nautilus-target.c <<EOF
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+void funcA(char* buf) {
+    buf[1] = 'A';
+}
+void funcB(char* buf) {
+    buf[2] = 'B';
+}
+
+int main(int argc, char** argv) {
+  char buf[1024];
+  int len = read(0, buf, 1024);
+  if (len < 0) return 1;
+  buf[len] = 0;
+  
+  if (buf[0] == 'A') {
+    for (int i = 0; i < 1000; i++) buf[i % 1024] ^= 1;
+  } else if (buf[0] == 'B') {
+    abort();
+  } else if (buf[0] == 'C') {
+    for (int i = 0; i < 500; i++) buf[i % 1024] ^= 2;
+  }
+  
+  if (strstr(buf, "test_custom_mutator") != NULL) {
+    abort();
+  }
+  return 0;
+}
+EOF
+
+  unset AFL_CC
+  # Compile the vulnerable program
+  rm -f test-nautilus-target
+  ../afl-cc -o test-nautilus-target test-nautilus-target.c > compilation_errors 2>&1
+
+
+  # Create input directory
+  mkdir -p in-nautilus
+  printf "AAAA" > in-nautilus/in
+
+  # Verify target compilation
+  if [ ! -x "./test-nautilus-target" ]; then
+    $ECHO "$RED[!] test-nautilus-target failed to compile."
+    echo "Compiler output:"
+    cat compilation_errors
+    rm -f compilation_errors
+    exit 1
+  fi
+  rm -f compilation_errors
+
+  # Run afl-fuzz with Nautilus mutator
+  # We set AFL_NO_UI=1 to avoid UI artifacts in logs
+  # We use -V 20 to run for 20 seconds (should be enough to crash)
+  $ECHO "$GREY[*] running afl-fuzz for the Nautilus mutator, this will take approx 20 seconds"
+  export NAUTILUS_GRAMMAR_FILE="$(pwd)/test-nautilus-grammar.json"
+  export AFL_NO_UI=1
+  export AFL_DISABLE_TRIM=1
+  # export AFL_CUSTOM_MUTATOR_ONLY=1
+  export AFL_MAP_SIZE=65536
+  export AFL_DRIVER_DONT_DEFER=1
+  export NAUTILUS_LOG=trace
+  
+  echo "DEBUG: Starting afl-fuzz..."
+  touch errors
+  AFL_CUSTOM_MUTATOR_LIBRARY="$LIB_PATH" timeout 60s ../afl-fuzz -V 20 -m ${MEM_LIMIT} -i in-nautilus -o out-nautilus -d -- ./test-nautilus-target >> errors 2>&1
+  echo "DEBUG: afl-fuzz finished with code $?"
+  ls -l errors
+  
+  unset NAUTILUS_GRAMMAR_FILE
+  unset AFL_NO_UI
+  unset AFL_CUSTOM_MUTATOR_ONLY
+
+  # Check if afl-fuzz ran successfully and loaded the mutator
+  if grep -F "Found 'afl_custom_mutator'" errors >/dev/null; then
```

Comment:
How does this work? the grammar does not contain the string that results in an abort in the harness